src/c-writer.cc

```diff
@@ -2471,15 +2471,19 @@ void CWriter::WriteImportProperties(CWriterPhase kind) {
   };
 
   for (const Import* import : unique_imports_) {
+    const Limits* limits = nullptr;
     if (import->kind() == ExternalKind::Memory) {
-      const Limits* memory_limits =
-          &(cast<MemoryImport>(import)->memory.page_limits);
-      write_import_prop(import, "min", memory_limits->initial);
-      write_import_prop(import, "max",
-                        memory_limits->has_max
-                            ? memory_limits->max
-                            : std::numeric_limits<uint32_t>::max());
+      limits = &(cast<MemoryImport>(import)->memory.page_limits);
+    } else if (import->kind() == ExternalKind::Table) {
+      limits = &(cast<TableImport>(import)->table.elem_limits);
+    } else {
+      continue;
     }
+
+    write_import_prop(import, "min", limits->initial);
+    write_import_prop(
+        import, "max",
+        limits->has_max ? limits->max : std::numeric_limits<uint32_t>::max());
```

Comment:
We may need to do something more complicated here because if `max` is unspecified, the implicit `max` depends on whether it's a table/memory32/memory64. I think it's:
- $2^{32} - 1$ elements (for a table)
- $2^{16}$ pages (for a memory.page_limits without `is_64` set)
- $2^{48}$ pages (for a memory.page_limits with `is_64` set)

(And, the types of the `min` and `max` variables probably should be u64 if limits.is_64 is set, and u32 otherwise.)

Reply:
(I wonder if it's easier just to not declare a `max` variable at all if the limits doesn't have a max? I.e. can your reflection system handle that situation?)

Reply:
hmm... the reflection can handle this, but it would add a lot of unnecessary hoops to jump through to make things work. From what I can see, the path of least resistance here would be 

- Make the min/max of memory a u64 and default to a max of 2^16 or 2^48 based on `is_64` . (Multiple types here will again make this harder to handle so I think sticking with u64 will be simpler)
- Keep the min/max of table a u32 and default to a max of 2^32 - 1

Does this seem reasonable?

Reply:
Works for me. At some point we may want to export an additional `is_64` variable because these are different types of memories (afaik an i32-indexed memory can't satisfy an import asking for an i64-indexed memory and vice versa, even if the max sizes are capped).